pkg/minikube/node/config.go

```diff
@@ -126,18 +129,13 @@ func setupKubeAdm(mAPI libmachine.API, cfg config.ClusterConfig, node config.Nod
 	return bs
 }
 
-func setupKubeconfig(h *host.Host, c *config.ClusterConfig, n *config.Node, clusterName string) (*kubeconfig.Settings, error) {
-	addr, err := h.Driver.GetURL()
+func setupKubeconfig(h *host.Host, cc *config.ClusterConfig, n *config.Node, clusterName string) (*kubeconfig.Settings, error) {
+	addr, err := apiServerURL(*h, *n)
 	if err != nil {
-		exit.WithError("Failed to get driver URL", err)
+		exit.WithError("Failed to get api server URL", err)
 	}
-	if !driver.IsKIC(h.DriverName) {
-		addr = strings.Replace(addr, "tcp://", "https://", -1)
-		addr = strings.Replace(addr, ":2376", ":"+strconv.Itoa(n.Port), -1)
-	}
-
-	if c.KubernetesConfig.APIServerName != constants.APIServerName {
-		addr = strings.Replace(addr, n.IP, c.KubernetesConfig.APIServerName, -1)
+	if cc.KubernetesConfig.APIServerName != constants.APIServerName {
+		addr = strings.Replace(addr, n.IP, cc.KubernetesConfig.APIServerName, -1)
```

Comment:
Now that the worst was refactored out, this search and replace is clearly broken, especially with KIC:

- The APIServerName is not guaranteed to be a hostname at all. It can be any arbitrary string.
- We're searching and replacing for an IP which may not exist in the string

I see this feature was added here: https://github.com/kubernetes/minikube/issues/3878

I don't totally love this feature, but my suggestion would be to move this logic to apiServerURL so that it can be done without search and replace. It probably isn't necessary to support with KIC, but the search and replace is currently broken in this use case, as n.IP == 172.17.0.x, and this URL uses 127.0.0.1.

Reply:
done